### PR TITLE
Unify use get hub

### PR DESCRIPTION
**Pull request by MilanPospisil**
Ensure useGet handles error correctly:
if (error) return <HubError/>
if (!data) return <LoadingPage/>

Change some of the useGetRequest to useGet.

Also refactor useCollectionSignatureUpload
 - remove one unecessary loading of data that are not used
 - use collection repo instead of hardcoded staging
 - correct the link to signatureUpload page

### CODE REVIEW

File: frontend/awx/common/AwxError.tsx [hunk 1]
```diff
@@ -10,13 +10,13 @@ import { ExclamationCircleIcon, SyncIcon } from '@patternfly/react-icons';
 import { useTranslation } from 'react-i18next';
 import { RequestError } from '../../common/crud/RequestError';
 
-export function AwxError(props: { error: Error; handleRefresh?: () => void }) {
+export function AwxError(props: { error: Error | undefined; handleRefresh?: () => void }) {
```
[the remark this answers is not in the document]
Comment (MilanPospisil) [reply frontend/awx/common/AwxError.tsx]: So far, I created Hub version that encapsulates AwxError, so I dont have to make my own decision in this PR how to create common error for both projects. This should be discussed.

**Review by ShaiahWren — COMMENTED** [review]
This looks correct, good changes, however there are some issues with the Approvals that I thought were going to be addressed in an earlier pr (for example, the approve and sign button does not actually sign it even though the flag for auto-signing is true). Would you like me to make some suggestions for that? I can also make a pr if that would be helpful.

**Comment by MilanPospisil** [comment]
> This looks correct, good changes, however there are some issues with the Approvals that I thought were going to be addressed in an earlier pr (for example, the approve and sign button does not actually sign it even though the flag for auto-signing is true). Would you like me to make some suggestions for that? I can also make a pr if that would be helpful.

Yes plase, you can do that if you wish :-)

**Review by nixocio — APPROVED** [review]
@MilanPospisil good job. 

We still have a couple of places, that we could add isLoading state and error handling, when using useGet, but not sure if the goal was to update the whole hub code basis.

**Pull request merged**